### PR TITLE
Mpuncel/aggregator improvement

### PR DESCRIPTION
**Pull request by mpuncel**

### CODE REVIEW

File: pkg/labels/consul_aggregator_test.go [hunk 1]
```diff
@@ -250,5 +250,80 @@ func TestCachedValueImmediatelySent(t *testing.T) {
...
+
+	select {
+	case <-time.After(time.Second):
+		t.Fatal("didn't get a value on the first selector")
+	case <-labeledChannel1:
+	}
+
+	select {
+	case <-time.After(time.Second):
+		t.Fatal("didn't get a value on the first selector")
```
Comment (rudle) [inline pkg/labels/consul_aggregator_test.go]: copy paste error: "second selector"

File: pkg/labels/consul_aggregator.go [hunk 2]
```diff
@@ -281,16 +282,27 @@ func (c *consulAggregator) sendMatches(watcher *selectorWatch) bool {
...
-	case <-watcher.resultCh:
-		sendSuccess = false
-	default:
-	}
-	// ... then send the newer value.
-	select {
-	case watcher.resultCh <- matches:
-	default:
+	var ret []bool
+	for watcher, _ := range watches.watches {
```
Comment (rudle) [inline pkg/labels/consul_aggregator.go]: can/should this be done in parallel?

Comment (mpuncel) [inline pkg/labels/consul_aggregator.go]: the channels are buffered so I don't think it matters much

File: pkg/labels/consul_aggregator.go [hunk 3]
```diff
@@ -22,19 +22,39 @@ type MetricsRegistry interface {
...
+	// is held, so no additonal mutex is necessary
+	watches map[selectorWatch]struct{}
+}
+
+type watchMap map[string]*selectorWatches
+
+func (w watchMap) len() int {
+	total := 0
+	for _, watches := range w {
+		for range watches.watches {
```
Comment (petertseng) [inline pkg/labels/consul_aggregator.go]: 🤔 could this be `total += len(watches.watches)`?

Comment (mpuncel) [inline pkg/labels/consul_aggregator.go]: yes 🤦‍♂️

**Pull request merged**